rest_api/actions/reviews.py

```diff
@@ -1,37 +1,63 @@
-import json
 import datetime
+import json
+
+from cached_property import cached_property
 from egcg_core import clarity
 from pyclarity_lims.entities import Queue, Step
 from requests.exceptions import HTTPError
 from werkzeug.exceptions import abort
 
+from config import rest_config as cfg
 from rest_api import settings
 from rest_api.aggregation.database_side import db
-from config import rest_config as cfg
 
 
-class ReviewInitiator:
+class Action(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @staticmethod
+    def now():
+        return datetime.datetime.now().strftime(settings.DATE_FORMAT)
+
+
+    @cached_property
+    def date_started(self):
+        return self.now()
+
+    def _perform_action(self):
+        raise NotImplementedError
+
+    def perform_action(self):
+        action_dict = {}
+        action_dict['date_started'] = self.date_started
+        if hasattr(self.request.authorization, 'username'):
+            action_dict['started_by'] = self.request.authorization.username
```

Comment:
'started_by' is present both here and in `_perform_action`. Is that because of the problems with token auth? If so, that can be cleared up in #105

Reply:
Not really. The automated processes do not required any extra authentication so only the logged in user is available. Here we have the username and password provided so we have a better idea of who started the step.